packages/venia-ui/lib/components/FilterModal/filterModal.js

```diff
@@ -66,9 +77,42 @@ const FilterModal = props => {
         </div>
     ) : null;
 
+    function handleKeyActions(event) {
+        if (isOpen) {
+            if (event.keyCode === 27) {
+                handleClose(event);
+            }
+            const element = modalRef.current;
+            const focusableEls = tabbable(element);
+            const firstFocusableEl = focusableEls[0];
+            const lastFocusableEl = focusableEls[focusableEls.length - 1];
+            if (event.keyCode === 9 || event.key === 'Tab') {
+                if (event.shiftKey) {
+                    /* shift + tab */
+                    if (document.activeElement === firstFocusableEl) {
+                        lastFocusableEl.focus();
+                        event.preventDefault();
+                    }
+                } /* tab */ else {
+                    if (document.activeElement === lastFocusableEl) {
+                        firstFocusableEl.focus();
+                        event.preventDefault();
+                    }
+                }
+            }
```

Comment:
is this logic can be converted to an util for other modals and dialogs

Reply:
@tjwiebell

Reply:
Let's not over optimize yet, but I do agree that this could be decomp'd into a general utility at some point.